test/hotspot/jtreg/compiler/vectorapi/reshape/utils/VectorReshapeHelper.java

```diff
@@ -107,32 +107,35 @@ public static void runMainHelper(Class<?> testClass, Stream<VectorSpeciesPair> t
         String testMethodName = VectorSpeciesPair.makePair(isp, osp).format();
         var caller = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).getCallerClass();
         var testMethod = MethodHandles.lookup().findStatic(caller,
-                testMethodName,
-                MethodType.methodType(void.class, byte.class.arrayType(), byte.class.arrayType()));
-        byte[] input = new byte[isp.vectorByteSize()];
-        byte[] output = new byte[osp.vectorByteSize()];
+                    testMethodName,
+                    MethodType.methodType(void.class, isp.elementType().arrayType(), osp.elementType().arrayType()))
+                .asType(MethodType.methodType(void.class, Object.class, Object.class));
+        Object input = Array.newInstance(isp.elementType(), isp.length());
+        Object output = Array.newInstance(osp.elementType(), osp.length());
+        long ibase = UnsafeUtils.arrayBase(isp.elementType());
+        long obase = UnsafeUtils.arrayBase(osp.elementType());
         for (int iter = 0; iter < INVOCATIONS; iter++) {
             // We need to generate arrays with NaN or very large values occasionally
             boolean normalArray = random.nextBoolean();
             var abnormalValue = List.of(Double.NaN, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, -1e30, 1e30);
             for (int i = 0; i < isp.length(); i++) {
                 switch (isp.elementType().getName()) {
-                    case "byte"   -> setByte(input, i, (byte)random.nextInt());
-                    case "short"  -> setShort(input, i, (short)random.nextInt());
-                    case "int"    -> setInt(input, i, random.nextInt());
-                    case "long"   -> setLong(input, i, random.nextLong());
+                    case "byte"   -> UnsafeUtils.putByte(input, ibase, i, (byte)random.nextInt());
```

Comment:
FWIW you could replace with `Array.getByte` etc, but those are non-intrinsic native methods, so that will likely slow things down. Another approach is to move the data setup to above the iteration loop, use a larger array, and move the data verification below the iteration loop. I think we should wait until we can use `MemorySegment`, then we can revisit if need be.